akka-docs/src/test/java/jdocs/typed/tutorial_4/DeviceGroup.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (C) 2009-2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package jdocs.typed.tutorial_4;
+
+import akka.actor.typed.ActorRef;
+import akka.actor.typed.Behavior;
+import akka.actor.typed.PostStop;
+import akka.actor.typed.javadsl.AbstractBehavior;
+import akka.actor.typed.javadsl.ActorContext;
+import akka.actor.typed.javadsl.Behaviors;
+import akka.actor.typed.javadsl.Receive;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static jdocs.typed.tutorial_4.DeviceManagerProtocol.*;
+import static jdocs.typed.tutorial_4.DeviceProtocol.DeviceMessage;
+
+//#device-group-full
+//#device-group-remove
+//#device-group-register
+public class DeviceGroup extends AbstractBehavior<DeviceGroupMessage> {
+
+  public static Behavior<DeviceGroupMessage> behavior(String groupId) {
+    return Behaviors.setup(context -> new DeviceGroup(context, groupId));
+  }
+
+  //#device-terminated
+  private class DeviceTerminated implements DeviceGroupMessage{
+    public final ActorRef<DeviceProtocol.DeviceMessage> device;
+    public final String groupId;
+    public final String deviceId;
+
+    DeviceTerminated(ActorRef<DeviceProtocol.DeviceMessage> device, String groupId, String deviceId) {
+      this.device = device;
+      this.groupId = groupId;
+      this.deviceId = deviceId;
+    }
+  }
+  //#device-terminated
+
+  private final ActorContext<DeviceGroupMessage> context;
+  private final String groupId;
+  private final Map<String, ActorRef<DeviceMessage>> deviceIdToActor = new HashMap<>();
+
+  public DeviceGroup(ActorContext<DeviceGroupMessage> context, String groupId) {
+    this.context = context;
+    this.groupId = groupId;
+    context.getLog().info("DeviceGroup {} started", groupId);
+  }
+
+  private DeviceGroup onTrackDevice(RequestTrackDevice trackMsg) {
+    if (this.groupId.equals(trackMsg.groupId)) {
+      ActorRef<DeviceMessage> deviceActor = deviceIdToActor.get(trackMsg.deviceId);
+      if (deviceActor != null) {
+        trackMsg.replyTo.tell(new DeviceRegistered(deviceActor));
```

Comment:
the original untyped version delegated the `RequestTrackDevice` down to the device actor, but I see no point in doing that